CMakeLists.txt

```diff
@@ -0,0 +1,106 @@
+cmake_minimum_required(VERSION 3.1)
+project(node-osrm C CXX)
+
+# Explicitly set the build type to Release if no other type is specified
+# on the command line.  Without this, cmake defaults to an unoptimized,
+# non-debug build, which almost nobody wants.
+if(NOT CMAKE_BUILD_TYPE)
+  message(STATUS "No build type specified, defaulting to Release")
+  set(CMAKE_BUILD_TYPE Release)
+endif()
+
+option(BUILD_LIBOSRM "Download and build own libsorm version" OFF)
+
+set(OSRM_BINARIES "")
+if(BUILD_LIBOSRM)
+  set(OSRM_INSTALL_DIR "/tmp/osrm-backend")
+  set(OSRM_DEPS_DIR "${CMAKE_CURRENT_SOURCE_DIR}/deps/osrm-backend-${CMAKE_BUILD_TYPE}")
+
+  execute_process(COMMAND node -p "require('../package.json').osrm_release" OUTPUT_VARIABLE OSRM_VERSION OUTPUT_STRIP_TRAILING_WHITESPACE)
+  message(STATUS "Using osrm ${OSRM_VERSION}")
+  message(STATUS "Downloading to ${OSRM_DEPS_DIR}")
+  message(STATUS "Installing to ${OSRM_INSTALL_DIR}")
+
+  execute_process(COMMAND git clone https://github.com/Project-OSRM/osrm-backend.git ${OSRM_DEPS_DIR})
+  execute_process(COMMAND git checkout ${OSRM_VERSION} WORKING_DIRECTORY ${OSRM_DEPS_DIR})
+
+  # These will be set by the subproject
+  # LibOSRM_INCLUDE_DIRS
+  # LibOSRM_LIBRARIES
+  # LibOSRM_DEPENDENT_LIBRARIES
+  # LibOSRM_CXXFLAGS
+  set(ENABLE_MASON ON CACHE INTERNAL "Always use mason for building osrm-backend" FORCE)
+  add_subdirectory(${OSRM_DEPS_DIR})
+  include(${OSRM_DEPS_DIR}/third_party/mason/mason.cmake)
+
+  # FIXME we can remove this double versioning after the merge
+  set(MASON_TBB_VERSION "43_20150316")
+  mason_use(tbb VERSION ${MASON_TBB_VERSION})
+
+  # We need this because we depend on the shared libraries from TBB
+  foreach(libpath ${MASON_PACKAGE_tbb_LIBRARY_DIRS})
+    file(GLOB TBBGlob ${libpath}/*.*)
+    foreach(filepath ${TBBGlob})
+      get_filename_component(filename ${filepath} NAME)
+      set(targetfile "${CMAKE_SOURCE_DIR}/lib/binding/${filename}")
+      add_custom_command(OUTPUT ${targetfile}
+                         COMMAND ${CMAKE_COMMAND} -E copy ${filepath} ${CMAKE_SOURCE_DIR}/lib/binding/
+                         DEPENDS ${filepath})
+      list(APPEND OSRM_BINARIES "${targetfile}")
+    endforeach()
+  endforeach()
```

Comment:
usually CMake will set a RPATH for every shared object (i.e. also executables) so they will find their dependencies from the current build tree, e.g. if you build both a shared library and an executable using it. These RPATH entries will be changed on "make install" to point to the final destination.